django_irods/views.py

```diff
@@ -0,0 +1,477 @@
+import datetime
+import json
+import mimetypes
+import os
+# import random
+from uuid import uuid4
+
+from django.conf import settings
+from django.core.exceptions import PermissionDenied
+from django.http import HttpResponse, FileResponse, HttpResponseRedirect
+from rest_framework.decorators import api_view
+
+from django_irods import icommands
+# from django_irods.storage import IrodsStorage
+from hs_core.hydroshare import check_resource_type
+from hs_core.hydroshare.hs_bagit import create_bag_files
+from hs_core.hydroshare.resource import FILE_SIZE_LIMIT
+from hs_core.signals import pre_download_file, pre_check_bag_flag
+from hs_core.tasks import create_bag_by_irods, create_temp_zip, delete_zip
+from hs_core.views.utils import authorize, ACTION_TO_AUTHORIZE
+from . import models as m
+from .icommands import Session, GLOBAL_SESSION
+from hs_core.models import ResourceFile
+
+import logging
+logger = logging.getLogger(__name__)
+
+
+def download(request, path, rest_call=False, use_async=True, use_reverse_proxy=True,
+             *args, **kwargs):
+    """ perform a download request, either asynchronously or synchronously
+
+    :param request: the request object.
+    :param path: the path of the thing to be downloaded.
+    :param rest_call: True if calling from REST API
+    :param use_async: True means to utilize asynchronous creation of objects to download.
+    :param use_reverse_proxy: True means to utilize NGINX reverse proxy for streaming.
+
+    The following variables are computed:
+
+    * `path` is the public path of the thing to be downloaded.
+    * `irods_path` is the location of `path` in irods.
+    * `output_path` is the output path to be reported in the response object.
+    * `irods_output_path` is the location of `output_path` in irods
+
+    and there are seven cases:
+
+    1. path points to a single file that is not a single-file-aggregation
+       object in a composite resource.
+    2. path points to a single file that is a single-file-aggregation
+       object in a composite resource.
+    3. path points to a metadata object that may need updating.
+    4. path points to a bag that needs to be updated and then returned.
+    5. path points to a folder that needs to be zipped.
+    6. path points to a previously zipped file that was zipped asynchronously.
+    7. path points to a .zip file that -- when the .zip is removed -- is a file that can be zipped
+
+    In cases 1, 3, 4, and 6 the output path is the input path.
+    In cases 2, 5, and 7, the output path is created as a result of
+    the operation of zipping the object and its metadata into a new zipfile.
+    """
+    logger.debug("request path is {}".format(path))
```

Comment:
Well, I would rather not because: 

1. We might need to debug this further. 
2. It has no effect except in debugging mode. 

When I am confident that everything is working, it can be removed.

Reply:
Then you need to add the condition to make sure this logger debug statement only happens when DEBUG is set to True, otherwise, this informational log will be written to the log in production as well